src/integrity-checker.js

```diff
@@ -33,30 +39,37 @@ export default class InstallationIntegrityChecker {
    * write a new one.
    */
 
-  async _getIntegrityHashLocation(): Promise<string> {
+  async _getIntegrityHashLocation(usedRegistries?: Set<RegistryNames>): Promise<IntegrityHashLocation> {
     // build up possible folders
+    let registries = registryNames;
+    if (usedRegistries && usedRegistries.size > 0) {
+      registries = usedRegistries;
+    }
     const possibleFolders = [];
     if (this.config.modulesFolder) {
       possibleFolders.push(this.config.modulesFolder);
     }
 
     // ensure we only write to a registry folder that was used
-    for (const name of registryNames) {
+    for (const name of registries) {
       const loc = path.join(this.config.cwd, this.config.registries[name].folder);
       possibleFolders.push(loc);
     }
 
     // if we already have an integrity hash in one of these folders then use it's location otherwise use the
     // first folder
     const possibles = possibleFolders.map((folder): string => path.join(folder, constants.INTEGRITY_FILENAME));
-    let loc = possibles[0];
+    let loc;
     for (const possibleLoc of possibles) {
       if (await fs.exists(possibleLoc)) {
         loc = possibleLoc;
         break;
       }
     }
-    return loc;
+    return {
+      locationPath: loc || possibles[0],
+      exists: !!loc,
+    };
```

Comment:
Not fond of the location getter that also check if a file exists, but it's a private method, so no big deal.